server/etcdserver/api/v3rpc/auth.go

```diff
@@ -164,3 +165,21 @@ func (as *AuthServer) UserChangePassword(ctx context.Context, r *pb.AuthUserChan
 	}
 	return resp, nil
 }
+
+type AuthGetter interface {
+	AuthInfoFromCtx(ctx context.Context) (*auth.AuthInfo, error)
+	AuthStore() auth.AuthStore
+}
+
+type AuthChecker struct {
+	ag AuthGetter
+}
+
+func (achk *AuthChecker) isAuthenticated(ctx context.Context) error {
```

Comment:
`Authentication` is not `Authorization`. 
```suggestion
func (achk *AuthChecker) isAuthorized(ctx context.Context) error {
```